### ISSUE TITLE
AMP timeago extension Persian (Farsi) locale

### ISSUE THREAD

**Issue by masoudhaghi** [ampproject/amphtml#30048]
## Describe the new feature or change to an existing feature you'd like to see

The timeago [timeagojs](https://github.com/ampproject/amphtml/tree/2fda798347d67df834fe2d14dd4ae617d511a4b7/third_party/timeagojs) extension of AMP is missing Persian (Farsi) locale `locale="fa"` which can be used in Persian and Farsi websites.

## Describe alternatives you've considered

Unfortunately as the extension files are hosted on amp project website, I'm not able to modify nor create custom script just to add a localization for Persian language. However, the following changes in the existing extension would be helpful in order to add the Persian localization for timeago extension.

## Additional context

Import and register the new locale in [timeago.js](third_party/timeagojs/timeago.js) file.

```JavaScript
import {fa} from './timeago-locales';
timeago.register('fa', fa);
```

Add the Persian locale function and strings in [timeago-locale.js](third_party/timeagojs/timeago-locales.js) file.

```JavaScript
/**
 * fa (Persian)
 */
export const fa = function(number, index) {
  return [
    ['همین الان', 'همین الان'],
    ['%s ثانیه پیش', 'در %s ثانیه آینده'],
    ['1 دقیقه پیش', 'در 1 دقیقه آینده'],
    ['%s دقیقه پیش', 'در %s دقیقه آینده'],
    ['1 ساعت پیش', 'در 1 ساعت آینده'],
    ['%s ساعت پیش', 'در %s ساعت آینده'],
    ['1 روز پیش', 'در 1 روز آینده'],
    ['%s روز پیش', 'در %s روز آینده'],
    ['1 هفته پیش', 'در 1 هفته آینده'],
    ['%s هفته پیش', 'در %s هفته آینده'],
    ['1 ماه پیش', 'در 1 ماه آینده'],
    ['%s ماه پیش', 'در %s ماه آینده'],
    ['1 سال پیش', 'در 1 سال آینده'],
    ['%s سال پیش', 'در %s سال آینده'],
  ][index];
};
```

**Comment by caroqliu** [comment]
@masoudhaghi Thanks for filing this! Would you be interested in making this contribution to `amp-timeago`? 😄

**Comment by masoudhaghi** [comment]
@caroqliu sure. I can also contribute to the localization of the AMP and the extensions if needed.

**Comment by michalmatuska** [comment]
Hi, what is the status here? Can you release new version of component with Farsi support? It will be blocker for my project. We are thinking about workaround but we don't like them if we have solution in front of us. Is anything what can I do?

**Comment by caroqliu** [comment]
@michalmatuska We have a fix it sprint coming up at the end of the month, and this issue would be a good one to tackle as part of that effort. If you need it sooner than that you are welcome to make the contribution, given the necessary content appears to be available in the original issue filed. Happy to be your reviewer and guide if so. :)

**Comment by caroqliu** [comment]
It looks like the timeago library has since [added support for Persian (Farsi)](https://github.com/hustcc/timeago.js/blob/master/src/lang/fa.ts), which we can use to add to `amp-timeago`. Thank you @masoudhaghi for your contribution of translations, though!

According to our `third_party` notes for the library, there were 37 locales at the time the component was contributed into AMPHTML:

https://github.com/ampproject/amphtml/blob/6549bb8a362e73a654f5e006f766882cbd74d710/third_party/timeagojs/README.amp#L9-L13

I propose updating our locales to include the [45 now supported](https://github.com/hustcc/timeago.js/tree/master/src/lang) in the timeago library. @rsimha Can you advise, since this is in our `third_party/` directory, if this update should be done by updating the version that we forked (to [4.0.2](https://github.com/hustcc/timeago.js/releases/tag/v4.0.2)) and reapplying the original modifications, or further modifying the current forked version (3.0.0) to include the new languages?

**Comment by rsimha** [comment]
> Can you advise, since this is in our `third_party/` directory, if this update should be done by updating the version that we forked (to [4.0.2](https://github.com/hustcc/timeago.js/releases/tag/v4.0.2)) and reapplying the original modifications, or further modifying the current forked version (3.0.0) to include the new languages?

Here are the preferred options, in order:
1. If the package can be used as-is with no modifications, add it to this list as a runtime dependency
  https://github.com/ampproject/amphtml/blob/6549bb8a362e73a654f5e006f766882cbd74d710/package.json#L16-L37
2. If the package can be used with minimal modifications, add it as a runtime dependency and patch it like we do to other packages
  https://github.com/ampproject/amphtml/blob/6549bb8a362e73a654f5e006f766882cbd74d710/build-system/tasks/update-packages.js#L185-L188
3. If the modifications are non-trivial, add it to `third_party/`, document the modifications, and keep the package up to date

I wonder if we can move from 3) to 2) given that we optimize JS to remove unused code using closure compiler. If the answer is no, I'd vote for updating to 4.0.2 to include all the new locales, and then freshly reapply the local modifications. It's a little more work, but will make for a more up-to-date and maintainable solution. WDYT?

**Comment by kristoferbaxter** [comment]
@caroqliu LIkely we need to get a diff from the original `third_party` fork, since this will continue to be a choice of either maintain or re-unite with original source.

My hunch is for the sake of moving more quickly, we should attempt to move to `4.0.2` in a timeboxed manner... if it's proving difficult after an hour or two, then see how long it would take to use the updated translations with our forked version.

**Comment by michalmatuska** [comment]
@caroqliu This is good news. How can I implement this change to my site?
When I am trying this https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fcomponents%2Famp-timeago.example.1.html%3Fformat%3Dwebsites&format=websites&_gl=1*5xuwzo*_ga*YW1wLUV6bW8td2VTd3FtX1ZfdlFfUjlZU0E. Farsi locale doesn't work.

**Comment by caroqliu** [comment]
The change will roll out according to the [AMP release schedule](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md):
> Once a change in AMP is merged into the master branch of the amphtml repository, it will typically take 1-2 weeks for the change to be live for all users.

In this case, I believe the change should be available with the March 9 release.

**Comment by michalmatuska** [comment]
@caroqliu thank you very much! I am looking forward.
